### PR TITLE
Specs2 toolchain

### PR DESCRIPTION
**Pull request by liucijus**
Specs2 toolchain, part of #940

### CODE REVIEW

File: test_version/WORKSPACE.template [hunk 1]
```diff
@@ -47,6 +47,8 @@ specs2_junit_repositories()
 
 register_toolchains("@io_bazel_rules_scala//testing:testing_toolchain")
 
+register_toolchains("@io_bazel_rules_scala//testing:testing_toolchain")
```
Comment (simuons) [inline test_version/WORKSPACE.template:50]: Duplicates line above.

File: testing/BUILD [hunk 2]
```diff
@@ -66,3 +85,24 @@ declare_deps_provider(
         "//external:io_bazel_rules_scala/dependency/scala/scalatest/scalatest",
     ],
 )
+
+declare_deps_provider(
+    name = "specs2_classpath_provider",
+    deps_id = "specs2_classpath",
+    visibility = ["//visibility:public"],
+    deps = [
+        "@io_bazel_rules_scala_org_specs2_specs2_common",
```
Comment (simuons) [inline testing/BUILD:94]: Why all other deps are in form `//external:io_bazel_rules_scala/dependency/...` and these are not?

File: test_version/WORKSPACE.template (hunk 1 above)
Comment (liucijus) [reply test_version/WORKSPACE.template:50]: Thanks for catching!

File: testing/BUILD (hunk 2 above)
Comment (liucijus) [reply testing/BUILD:94]: It comes from the existing code: https://github.com/bazelbuild/rules_scala/blob/master/specs2/BUILD#L9

It's binded later in https://github.com/bazelbuild/rules_scala/blob/master/specs2/specs2.bzl#L26

Existing binds are subject to be removed later. Currently the goal is to give a way to not use binds.

File: docs/testing.md [hunk 3]
```diff
@@ -11,12 +11,17 @@ junit_repositories()
...
 scalatest_toolchain()
+
+# Specs2 with Junit
+load("@io_bazel_rules_scala//testing:specs2_junit.bzl", "specs2_junit_repositories", "specs2_junit_toolchain")
+specs2_junit_repositories()
+specs2_junit_toolchain()
 ```
 
-### Example to set up JUnit dependencies
+### Configuring JUnit dependencies vai toolchain
```
Comment (ittaiz) [inline docs/testing.md:24]: nit: typo

**Review by ittaiz — APPROVED** [review]

File: docs/testing.md (hunk 3 above)
Comment (liucijus) [reply docs/testing.md]: thanks for catching - fixed

**Review by ittaiz — APPROVED** [review]
will merge once green. thanks!

**Pull request merged**